src/string-calculator.ts

```diff
@@ -1,12 +1,235 @@
+            this.lastErrorCode = ErrorCode.CustomSeparatorFoundErrorCode;
+            return this.errorMessage;
+        }],
+        [ErrorCode.EndOfLineFoundErrorCode, (): string => {
+            let position: number = this.getIndexOfUnexpectedEndOfLine();
+            if (this.errorMessage.trim().length != 0) {
+                this.errorMessage = this.errorMessage.concat('\n');
+            }
+            this.errorMessage = this.errorMessage.concat(`Number expected but \'\\n\' found at position ${position}.`);
+            this.lastErrorCode = ErrorCode.EndOfLineFoundErrorCode;
+            return this.errorMessage;
+        }],
+        [ErrorCode.LastNumberMissedErrorCode, (): string => {
+            if (!this.isUnexpectedEOLAdded) {
+                if (this.errorMessage.trim().length != 0) {
+                    this.errorMessage = this.errorMessage.concat('\n');
+                }
+                this.errorMessage = this.errorMessage.concat(`Number expected but EOF found.`);
+                this.isUnexpectedEOLAdded = true;
+            }
+            this.lastErrorCode = ErrorCode.LastNumberMissedErrorCode;
+            return this.errorMessage;
+        }],
+        [ErrorCode.NegativeNumberErrorCode, (number: string): string => {
+            if (this.lastErrorCode === ErrorCode.NegativeNumberErrorCode) {
+                this.errorMessage = this.errorMessage.concat(", " + number);
+            } else {
+                if (this.errorMessage.trim().length != 0) {
+                    this.errorMessage = this.errorMessage.concat('\n');
+                }
+                this.errorMessage = this.errorMessage.concat(`Negative not allowed : ` + number);
+            }
+            this.lastErrorCode = ErrorCode.NegativeNumberErrorCode;
+            return this.errorMessage;
+        }],
+        [ErrorCode.NumberNotNumericErrorCode, (number: string): string => {
+            let unexpectedChar = this.getUnexpectedChar(number);
+            let position = this.numberList.indexOf(unexpectedChar);
+
+            if (this.errorMessage.trim().length != 0) {
+                this.errorMessage = this.errorMessage.concat('\n');
+            }
+            this.errorMessage = this.errorMessage.concat(`'${this.customSep}' expected but '${unexpectedChar}' found at position ${position}.`)
+
+            this.lastErrorCode = ErrorCode.NumberNotNumericErrorCode;
+            return this.errorMessage;
+        }]
+    ]);
+
+    public summation(numberList: string): string {
+        this.numberList = numberList;
+
+        if (this.numberList.length === 0) {
             return "0";
         }
 
+        let result: number = 0;
+        let regexToSplitListIntoNumbers = this.createRegex();
+        let listOfNumbers: string[] = this.numberList.split(new RegExp(regexToSplitListIntoNumbers));
+
+        listOfNumbers.forEach(number => {
+
+            let err: ErrorCode = this.checkForErrors(number);
```

Comment:
just a reminder: we talked about throwing error with a property error code so that we skip the 'if' part